com.microsoft.mrtk.spatialmanipulation/ObjectManipulator/NewObjectManipulator.cs

```diff
@@ -0,0 +1,572 @@
+            set => rotationAnchorNear = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Rotation behavior of object when using one hand at distance")]
+        private RotateAnchorType rotationAnchorFar = RotateAnchorType.RotateAboutGrabPoint;
+
+        /// <summary>
+        /// Rotation behavior of object when using one hand at distance
+        /// </summary>
+        public RotateAnchorType RotationAnchorFar
+        {
+            get => rotationAnchorFar;
+            set => rotationAnchorFar = value;
+        }
+
+        [SerializeField]
+        [EnumFlags]
+        [Tooltip("Rigid body behavior of the dragged object when releasing it.")]
+        private ReleaseBehaviorType releaseBehavior = ReleaseBehaviorType.KeepVelocity | ReleaseBehaviorType.KeepAngularVelocity;
+
+        /// <summary>
+        /// Rigid body behavior of the dragged object when releasing it.
+        /// </summary>
+        public ReleaseBehaviorType ReleaseBehavior
+        {
+            get => releaseBehavior;
+            set => releaseBehavior = value;
+        }
+
+        [FormerlySerializedAs("smoothingActive")]
+        [SerializeField]
+        [Tooltip("Frame-rate independent smoothing for far interactions. Far smoothing is enabled by default.")]
+        private bool smoothingFar = true;
+
+        /// <summary>
+        /// Whether to enable frame-rate independent smoothing for far interactions.
+        /// </summary>
+        /// <remarks>
+        /// Far smoothing is enabled by default.
+        /// </remarks>
+        public bool SmoothingFar
+        {
+            get => smoothingFar;
+            set => smoothingFar = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Frame-rate independent smoothing for near interactions. Note that enabling near smoothing may be perceived as being 'disconnected' from the hand.")]
+        private bool smoothingNear = true;
+
+        /// <summary>
+        /// Whether to enable frame-rate independent smoothing for near interactions.
+        /// </summary>
+        /// <remarks>
+        /// Note that enabling near smoothing may be perceived as being 'disconnected' from the hand.
+        /// </remarks>
+        public bool SmoothingNear
+        {
+            get => smoothingNear;
+            set => smoothingNear = value;
+        }
+
+        [Serializable]
+        /// <summary>
+        /// The SystemTypes for the desired type of manipulation logic for move, rotate, and scale.
+        /// </summary>
+        public struct LogicType
+        {
+            [SerializeField]
+            [Tooltip("The concrete type of ManipulationLogic<Vector3> to use for moving.")]
+            [Extends(typeof(ManipulationLogic<Vector3>), TypeGrouping.ByNamespaceFlat)]
+            /// <summary>
+            /// The concrete type of <see cref="ManipulationLogic"/> to use for moving.
+            /// </summary>
+            public SystemType moveLogicType;
```

Comment:
I wonder if these could use @maluoi's new (to MRTK at least) pattern of `[SerializeReference, InterfaceSelector]`. I know this extends a non-interface, but _I think_ that pattern should still work? Would save us the hassle of the `Activator` stuff too, potentially?

Reply:
Because ManipulationLogic is a [Generic Inflated Type](https://forum.unity.com/threads/serializereference-generic-inflated-types.916094/), it cannot be serialized with this attribute unfortunately :(